expander.go

```diff
@@ -369,6 +378,19 @@ func normalizePaths(refPath, base string) string {
 	return baseURL.String()
 }
 
+// denormalizePaths returns to simplest notation on file $ref,
+// i.e. strips the absolute path and sets a path relative to the base path.
+//
+// This is currently used when we rewrite ref after a circular ref has been detected
+func denormalizeFileRef(ref *Ref, relativeBase string) *Ref {
+	if ref.String() == "" || ref.IsRoot() || ref.HasFragmentOnly {
+		return ref
+	}
+	// strip relativeBase from URI
+	r, _ := NewRef(strings.TrimPrefix(ref.String(), relativeBase))
```

Comment:
This line is causing the issue in the validation test. It is creating an invalid JSON Ref, if I change it to this:

```
  // strip relativeBase from URI
	relativeBaseURL, _ := url.Parse(relativeBase)
	relativeBaseURL.Fragment = ""
	r, _ := NewRef(strings.TrimPrefix(ref.String(), relativeBaseURL.String()))
```

Then all of the test in the validate package pass except one. It should be really easy to fix, I have to run, but will try to get to it later if someone else hasn't.